Fix minor typos found during familiarization (#4556)
nasa/fprime: docs/user-manual/index.md

- __FPP User's Guide__ - In-depth user guide for F Prime Prime (FPP), the F´ modeling language.

> - __F´ GDS__ - Learn how to use a sample Ground Data System, and how it can be used to test F´ applications.

Reviewer: Why was it called "a sample Ground Data System"

Author: My thought was to indicate that other GDS's also work; it's not like we lock FPrime users into this one.  I may be unclear on the ambitions for F' GDS as its own product.

> - __F´ GDS__ - Learn how to use a sample Ground Data System, and how it can be used to test F´ applications.

Reviewer 2: ```suggestion
- __F´ GDS__ - Learn how to use the F Prime Ground Data System, and how it can be used to test F´ applications.
```
